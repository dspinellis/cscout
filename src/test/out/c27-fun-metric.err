Processing workspace TestWS
Processing directory test/c
Entering directory test/c
Processing project Prj1
Processing file c27-fun-metric.c
Done processing file c27-fun-metric.c
Done processing project Prj1
Processing project Prj2
Processing file c27-fun-metric.c
Done processing file c27-fun-metric.c
Processing file prj2.c
Done processing file prj2.c
Done processing project Prj2
Exiting directory test/c
Done processing directory test/c
Done processing workspace TestWS
Post-processing /cscout/example/.cscout/host-defs.h
Post-processing /cscout/example/.cscout/host-incs.h
Post-processing /cscout/src/makecs.cs
Post-processing /cscout/src/test/c/c27-fun-metric.c
Post-processing /cscout/src/test/c/prj2.c
Processing identifiers
4%9%13%18%22%27%31%36%40%45%50%54%59%63%68%72%77%81%86%90%95%100%
