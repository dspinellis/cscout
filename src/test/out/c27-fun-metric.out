Table: Ids
EID	NAME	READONLY	UNDEFMACRO	MACRO	FUNMACRO	MACROARG	CPPCONST	CPPSTRVAL	DEFCCONSTVAL	NOTDEFCCONSTVAL	EXPCCONSTVAL	NOTEXPCCONSTVAL	ORDINARY	SUETAG	SUMEMBER	LABEL	TYPEDEF	ENUM	YACC	FUN	CSCOPE	LSCOPE	UNUSED
91	MAXDIGIT	0	0	1	0	0	0	0	1	0	0	0	0	0	0	0	0	0	0	0	0	0	1
321	x	0	0	0	0	0	0	0	0	0	0	0	1	0	0	0	0	0	0	0	0	1	1
421	main	1	0	0	0	0	0	0	0	0	0	0	1	0	0	0	0	0	0	1	0	1	0
439	fun	0	0	0	0	0	0	0	0	0	0	0	1	0	0	0	0	0	0	1	0	1	1
541	qqq	0	0	0	0	0	0	0	0	0	0	0	1	0	0	0	0	0	0	0	0	1	1
609	decl1	0	0	0	0	0	0	0	0	0	0	0	1	0	0	0	0	0	0	0	0	0	0
759	sqr	0	0	1	1	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	1
799	x	0	0	0	0	1	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0
821	foo	0	0	0	0	0	0	0	0	0	0	0	1	0	0	0	0	0	0	1	0	1	1
959	decl2	0	0	0	0	0	0	0	0	0	0	0	1	0	0	0	0	0	0	0	0	0	0
2069	foo	0	0	1	1	0	0	0	0	0	0	1	0	0	0	0	0	0	0	0	0	0	0
2339	bar	0	0	1	1	0	0	0	0	0	0	1	0	0	0	0	0	0	0	0	0	0	0
2975	__DATE__	1	0	1	0	0	0	0	0	1	0	0	0	0	0	0	0	0	0	0	0	0	1
3295	__TIME__	1	0	1	0	0	0	0	0	1	0	0	0	0	0	0	0	0	0	0	0	0	1
3575	__FILE__	1	0	1	0	0	0	0	0	1	0	0	0	0	0	0	0	0	0	0	0	0	1
3865	__LINE__	1	0	1	0	0	0	0	1	0	0	0	0	0	0	0	0	0	0	0	0	0	1
4055	__STDC__	1	0	1	0	0	0	0	1	0	0	0	0	0	0	0	0	0	0	0	0	0	1
4169	many_compiled	0	0	0	0	0	0	0	0	0	0	0	1	0	0	0	0	0	0	1	0	1	1
4399	i	0	0	0	0	0	0	0	0	0	0	0	1	0	0	0	0	0	0	0	0	0	1
5355	_cscout_dummy1	1	0	0	0	0	0	0	0	0	0	0	1	0	0	0	0	0	0	1	1	0	0
Table: Tokens
FID	FOFFSET	EID
2	297	2975
2	329	3295
2	357	3575
2	386	3865
2	405	4055
2	471	421
2	535	5355
2	558	5355
4	43	439
4	60	609
4	75	759
4	79	799
4	83	799
4	87	799
4	95	959
4	111	959
4	183	609
4	206	2069
4	233	2339
4	263	2339
4	343	2069
4	351	2339
4	416	4169
4	439	4399
4	443	2339
5	8	91
5	31	321
5	41	421
5	53	541
5	81	821
Table: Rest
FID	FOFFSET	CODE
2	287	\u0000a\u0000a#define 
2	305	  
2	320	\u0000a#define 
2	337	 
2	348	\u0000a#define 
2	365	 
2	377	\u0000a#define 
2	394	 1\u0000a#define 
2	413	 1\u0000a\u0000a
2	466	\u0000aint 
2	475	();\u0000a
2	522	\u0000astatic void 
2	549	(void) { 
2	572	(); }\u0000a
3	152	\u0000a\u0000a\u0000a#pragma includepath 
3	195	\u0000a
3	239	\u0000astatic void _cscout_dummy2(void) { _cscout_dummy2(); }\u0000a
4	36	\u0000a\u0000avoid\u0000a
4	46	(void)\u0000a{\u0000a	int 
4	65	;\u0000a#define 
4	78	(
4	80	) (
4	84	 * 
4	88	)\u0000a	int 
4	100	;\u0000a	switch (
4	116	) {\u0000a#if 1\u0000a	case 1:\u0000a#elif 2\u0000a	case 3:\u0000a#endif\u0000a	case 'a':\u0000a\u0000a	default:\u0000a		
4	188	 = 12;\u0000a	}\u0000a#define 
4	209	() do { \\u0000a	for (;;) \\u0000a		
4	236	(); \\u0000a	} while (0)\u0000a#define 
4	266	() if (0) if (0) if (0) if (0) if (0) if (0) if (0) if (0) if (0) if (0) 1\u0000a\u0000a	
4	346	();\u0000a	
4	354	();\u0000a	if (0)\u0000a		if (0)\u0000a			1;\u0000a		else\u0000a			while (1)\u0000a				2;\u0000a}\u0000a\u0000avoid\u0000a
4	429	()\u0000a{\u0000a	int 
4	440	;\u0000a	
4	446	();\u0000a}\u0000a
5	0	#define 
5	16	 11\u0000aextern int 
5	32	;\u0000aextern 
5	45	();\u0000aint 
5	56	;\u0000a\u0000a
5	80	\u0000a
5	84	() {\u0000a	
5	107	}\u0000a\u0000a
Table: Projects
PID	NAME
23	unspecified
24	Prj1
25	Prj2
Table: IdProj
EID	PID
421	24
439	24
609	24
759	24
799	24
959	24
2069	24
2339	24
2975	24
3295	24
3575	24
3865	24
4055	24
4169	24
4399	24
5355	24
91	25
321	25
421	25
439	25
541	25
609	25
759	25
799	25
821	25
959	25
2069	25
2339	25
2975	25
3295	25
3575	25
3865	25
4055	25
4169	25
4399	25
5355	25
Table: Files
FID	NAME	RO
2	host-defs.h	1
3	host-incs.h	1
4	c27-fun-metric.c	0
5	prj2.c	0
Table: Filemetrics
FID	PRECPP	NCHAR	NCCOMMENT	NSPACE	NLCOMMENT	NBCOMMENT	NLINE	MAXLINELEN	MAXSTMTLEN	MAXSTMTNEST	MAXBRACENEST	MAXBRACKNEST	BRACENEST	BRACKNEST	NULINE	NPPDIRECTIVE	NPPCOND	NPPFMACRO	NPPOMACRO	NTOKEN	NSTMT	NOP	NUOP	NNCONST	NCLIT	NSTRING	NPPCONCATOP	NPPSTRINGOP	NIF	NELSE	NSWITCH	NCASE	NDEFAULT	NBREAK	NFOR	NWHILE	NDO	NCONTINUE	NGOTO	NRETURN	NASM	NTYPEOF	NPID	NFID	NMID	NID	NUPID	NUFID	NUMID	NUID	NLABEL	NMACROEXPANDTOKEN	NCOPIES	NINCFILE	NPFUNCTION	NFFUNCTION	NPVAR	NFVAR	NAGGREGATE	NAMEMBER	NENUM	NEMEMBER
2	0	NULL	NULL	NULL	NULL	NULL	NULL	NULL	16	0	1	1	NULL	NULL	NULL	NULL	NULL	NULL	NULL	18	2	0	0	0	0	0	NULL	NULL	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	NULL	0	0	0	NULL	0	0	NULL	NULL	NULL	0	1	0	0	0	0	0	0
2	1	548	367	29	0	3	22	61	47	NULL	1	1	0	0	0	5	0	0	5	37	2	0	0	2	0	3	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	1	2	5	8	1	1	5	7	0	0	1	0	NULL	NULL	NULL	NULL	NULL	NULL	NULL	NULL
3	0	NULL	NULL	NULL	NULL	NULL	NULL	NULL	15	0	1	1	NULL	NULL	NULL	NULL	NULL	NULL	NULL	0	1	0	0	0	0	0	NULL	NULL	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	NULL	0	0	0	NULL	0	0	NULL	NULL	NULL	0	0	0	0	0	0	0	0
3	1	243	187	13	0	2	13	48	28	NULL	1	1	0	0	0	1	0	0	0	16	1	0	0	0	0	1	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	1	0	NULL	NULL	NULL	NULL	NULL	NULL	NULL	NULL
4	0	NULL	NULL	NULL	NULL	NULL	NULL	NULL	64	2	2	1	NULL	NULL	NULL	NULL	NULL	NULL	NULL	197	47	1	0	41	1	0	NULL	NULL	32	1	1	2	1	0	1	1	1	0	0	0	0	0	4	0	NULL	14	2	0	NULL	5	0	NULL	NULL	NULL	2	0	0	0	0	0	0	0
4	1	384	32	124	0	1	41	71	84	NULL	2	1	0	0	3	6	2	3	0	158	28	2	0	22	1	0	0	0	13	1	1	3	1	0	1	1	1	0	0	0	0	0	2	0	7	14	2	0	3	8	0	400	1	0	NULL	NULL	NULL	NULL	NULL	NULL	NULL	NULL
5	0	NULL	NULL	NULL	NULL	NULL	NULL	NULL	7	0	1	1	NULL	NULL	NULL	NULL	NULL	NULL	NULL	18	3	0	0	0	0	0	NULL	NULL	0	0	0	0	0	0	0	0	0	0	0	0	0	0	4	0	NULL	4	4	0	NULL	4	0	NULL	NULL	NULL	1	0	2	0	0	0	0	0
5	1	95	32	17	2	0	11	20	13	NULL	1	1	0	0	0	1	0	0	1	21	3	0	0	1	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	4	0	1	5	4	0	1	5	0	0	1	0	NULL	NULL	NULL	NULL	NULL	NULL	NULL	NULL
Table: FileProj
FID	PID
2	24
3	24
4	24
1	25
2	25
3	25
4	25
5	25
Table: LineProj
FID	PID	LNUM
2	24	1
2	24	2
2	24	3
2	24	4
2	24	5
2	24	6
2	24	7
2	24	8
2	24	9
2	24	10
2	24	11
2	24	12
2	24	13
2	24	14
2	24	15
2	24	16
2	24	17
2	24	18
2	24	19
2	24	20
2	24	21
2	24	22
3	24	1
3	24	2
3	24	3
3	24	4
3	24	5
3	24	6
3	24	7
3	24	8
3	24	9
3	24	10
3	24	11
3	24	12
3	24	13
4	24	1
4	24	2
4	24	3
4	24	4
4	24	5
4	24	6
4	24	7
4	24	8
4	24	9
4	24	10
4	24	11
4	24	15
4	24	16
4	24	17
4	24	18
4	24	19
4	24	20
4	24	21
4	24	22
4	24	23
4	24	24
4	24	25
4	24	26
4	24	27
4	24	28
4	24	29
4	24	30
4	24	31
4	24	32
4	24	33
4	24	34
4	24	35
4	24	36
4	24	37
4	24	38
4	24	39
4	24	40
4	24	41
2	25	1
2	25	2
2	25	3
2	25	4
2	25	5
2	25	6
2	25	7
2	25	8
2	25	9
2	25	10
2	25	11
2	25	12
2	25	13
2	25	14
2	25	15
2	25	16
2	25	17
2	25	18
2	25	19
2	25	20
2	25	21
2	25	22
3	25	1
3	25	2
3	25	3
3	25	4
3	25	5
3	25	6
3	25	7
3	25	8
3	25	9
3	25	10
3	25	11
3	25	12
3	25	13
4	25	1
4	25	2
4	25	3
4	25	4
4	25	5
4	25	6
4	25	7
4	25	8
4	25	9
4	25	10
4	25	11
4	25	15
4	25	16
4	25	17
4	25	18
4	25	19
4	25	20
4	25	21
4	25	22
4	25	23
4	25	24
4	25	25
4	25	26
4	25	27
4	25	28
4	25	29
4	25	30
4	25	31
4	25	32
4	25	33
4	25	34
4	25	35
4	25	36
4	25	37
4	25	38
4	25	39
4	25	40
4	25	41
5	25	1
5	25	2
5	25	3
5	25	4
5	25	5
5	25	6
5	25	7
5	25	8
5	25	9
5	25	10
5	25	11
Table: LinePos
FID	FOFFSET	LNUM
2	0	1
2	4	2
2	51	3
2	54	4
2	100	5
2	119	6
2	122	7
2	184	8
2	231	9
2	234	10
2	284	11
2	288	12
2	289	13
2	321	14
2	349	15
2	378	16
2	397	17
2	416	18
2	417	19
2	467	20
2	479	21
2	523	22
3	0	1
3	4	2
3	51	3
3	54	4
3	78	5
3	97	6
3	100	7
3	149	8
3	153	9
3	154	10
3	155	11
3	196	12
3	240	13
4	0	1
4	37	2
4	38	3
4	43	4
4	53	5
4	55	6
4	67	7
4	90	8
4	102	9
4	120	10
4	126	11
4	135	12
4	143	13
4	152	14
4	159	15
4	170	16
4	171	17
4	181	18
4	195	19
4	198	20
4	219	21
4	231	22
4	242	23
4	255	24
4	341	25
4	342	26
4	350	27
4	358	28
4	366	29
4	375	30
4	381	31
4	388	32
4	401	33
4	408	34
4	410	35
4	411	36
4	416	37
4	432	38
4	434	39
4	442	40
4	450	41
5	0	1
5	20	2
5	34	3
5	49	4
5	58	5
5	59	6
5	80	7
5	81	8
5	89	9
5	107	10
5	109	11
Table: Definers
PID	CUID	BASEFILEID	DEFINERID
25	5	5	2
Table: Includers
PID	CUID	BASEFILEID	INCLUDERID
24	2	2	1
24	4	3	1
24	4	4	1
25	2	2	1
25	2	2	1
25	4	3	1
25	4	4	1
25	5	3	1
25	5	5	1
Table: Providers
PID	CUID	PROVIDERID
24	2	2
24	4	4
25	2	2
25	2	2
25	4	4
25	5	5
Table: IncTriggers
PID	CUID	BASEFILEID	DEFINERID	FOFFSET	LEN
25	5	5	2	471	4
Table: Functions
ID	NAME	ISMACRO	DEFINED	DECLARED	FILESCOPED	FID	FOFFSET	FANIN
439	fun	0	1	1	0	4	43	0
759	sqr	1	1	0	1	4	75	0
821	foo	0	1	1	0	5	81	0
2069	foo	1	1	0	1	4	206	1
2639	bar	1	1	0	1	4	263	3
4169	many_compiled	0	1	1	0	4	416	0
4715	main	0	0	1	0	2	471	0
5355	_cscout_dummy1	0	1	1	1	2	535	1
Table: FunctionDefs
FUNCTIONID	FIDBEGIN	FOFFSETBEGIN	FIDEND	FOFFSETEND
439	4	54	4	409
759	4	81	4	90
821	5	88	5	108
2069	4	211	4	255
2639	4	268	4	341
4169	4	433	4	451
5355	2	557	2	577
Table: FunctionMetrics
FUNCTIONID	PRECPP	NCHAR	NCCOMMENT	NSPACE	NLCOMMENT	NBCOMMENT	NLINE	MAXLINELEN	MAXSTMTLEN	MAXSTMTNEST	MAXBRACENEST	MAXBRACKNEST	BRACENEST	BRACKNEST	NULINE	NPPDIRECTIVE	NPPCOND	NPPFMACRO	NPPOMACRO	NTOKEN	NSTMT	NOP	NUOP	NNCONST	NCLIT	NSTRING	NPPCONCATOP	NPPSTRINGOP	NIF	NELSE	NSWITCH	NCASE	NDEFAULT	NBREAK	NFOR	NWHILE	NDO	NCONTINUE	NGOTO	NRETURN	NASM	NTYPEOF	NPID	NFID	NMID	NID	NUPID	NUFID	NUMID	NUID	NLABEL	NMACROEXPANDTOKEN	NGNSOC	NMPARAM	NFPARAM	NEPARAM	FANIN	FANOUT	CCYCL1	CCYCL2	CCYCL3	CSTRUC	CHAL	IFLOW
439	0	NULL	NULL	NULL	NULL	NULL	NULL	NULL	64	2	1	1	NULL	NULL	NULL	NULL	NULL	NULL	NULL	139	35	1	1	30	1	0	NULL	NULL	22	1	1	2	1	0	1	1	1	0	0	0	0	0	0	0	NULL	4	0	0	NULL	2	0	NULL	11	NULL	0	NULL	0	2	27	27	28	0.0	183.149	0.0
439	1	180	0	66	0	0	23	25	53	NULL	1	1	0	0	0	6	2	3	0	77	12	1	1	10	1	0	0	0	3	1	1	3	1	0	0	1	0	0	0	0	0	0	0	0	5	9	0	0	3	5	0	140	NULL	0	NULL	0	0	2	6	6	8	0.0	85.8367	0.0
759	0	NULL	NULL	NULL	NULL	NULL	NULL	NULL	0	0	0	0	NULL	NULL	NULL	NULL	NULL	NULL	NULL	0	0	0	0	0	0	0	NULL	NULL	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	NULL	0	0	0	NULL	0	0	NULL	0	NULL	0	NULL	0	0	1	1	1	0.0	0.0	0.0
759	1	12	0	5	0	0	1	10	0	NULL	0	1	0	0	0	0	0	0	0	5	0	1	1	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	NULL	1	NULL	0	0	0	1	1	1	0.0	0.0	0.0
821	0	NULL	NULL	NULL	NULL	NULL	NULL	NULL	0	0	0	0	NULL	NULL	NULL	NULL	NULL	NULL	NULL	1	0	0	0	0	0	0	NULL	NULL	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	NULL	0	0	0	NULL	0	0	NULL	18	NULL	0	NULL	0	0	1	1	1	0.0	0.0	0.0
821	1	21	14	3	1	0	3	17	0	NULL	0	0	0	0	0	0	0	0	0	1	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	NULL	0	NULL	0	0	0	1	1	1	0.0	0.0	0.0
2069	0	NULL	NULL	NULL	NULL	NULL	NULL	NULL	0	0	0	0	NULL	NULL	NULL	NULL	NULL	NULL	NULL	0	0	0	0	0	0	0	NULL	NULL	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	NULL	0	0	0	NULL	0	0	NULL	0	NULL	0	NULL	1	1	1	1	1	1.0	0.0	1.0
2069	1	39	0	16	0	0	4	11	9	NULL	1	1	0	0	0	0	0	0	0	16	4	0	0	1	0	0	0	0	0	0	0	0	0	0	1	0	1	0	0	0	0	0	0	0	1	1	0	0	1	1	0	0	NULL	0	NULL	0	1	1	3	3	3	1.0	2.0	3.0
2639	0	NULL	NULL	NULL	NULL	NULL	NULL	NULL	0	0	0	0	NULL	NULL	NULL	NULL	NULL	NULL	NULL	0	0	0	0	0	0	0	NULL	NULL	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	NULL	0	0	0	NULL	0	0	NULL	0	NULL	0	NULL	3	0	1	1	1	0.0	0.0	0.0
2639	1	64	0	23	0	0	2	62	0	NULL	0	1	0	0	0	0	0	0	0	41	10	0	0	11	0	0	0	0	10	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	NULL	0	NULL	0	3	0	11	11	11	0.0	38.0537	0.0
4169	0	NULL	NULL	NULL	NULL	NULL	NULL	NULL	63	0	0	1	NULL	NULL	NULL	NULL	NULL	NULL	NULL	46	12	0	0	11	0	0	NULL	NULL	10	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	NULL	1	0	0	NULL	1	0	NULL	16	NULL	0	NULL	0	1	11	11	11	0.0	43.0196	0.0
4169	1	19	0	7	0	0	4	8	6	NULL	0	1	0	0	0	0	0	0	0	8	2	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	1	2	0	0	1	2	0	60	NULL	0	NULL	0	0	1	1	1	1	0.0	2.0	0.0
5355	0	NULL	NULL	NULL	NULL	NULL	NULL	NULL	5	0	0	1	NULL	NULL	NULL	NULL	NULL	NULL	NULL	5	1	0	0	0	0	0	NULL	NULL	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	1	NULL	1	0	1	NULL	1	0	NULL	10	NULL	0	NULL	1	1	1	1	1	1.0	0.0	1.0
5355	1	22	0	3	0	0	1	21	5	NULL	0	1	0	0	0	0	0	0	0	5	1	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	1	0	1	0	1	0	1	0	0	NULL	0	NULL	0	1	1	1	1	1	1.0	0.0	1.0
Table: FunctionId
FUNCTIONID	ORDINAL	FID	FOFFSET	LEN
439	0	4	43	3
759	0	4	75	3
821	0	5	81	3
2069	0	4	206	3
2639	0	4	263	3
4169	0	4	416	13
4715	0	2	471	4
5355	0	2	535	14
Table: Fcalls
SOURCEID	DESTID
439	2069
439	2639
2069	2639
4169	2639
5355	5355
Done
